### PR TITLE
Accept a Donor's phone number for SMS notifications

### PR DESCRIPTION
**Pull request by seanpdoyle**
https://trello.com/c/Qumm2XI6

Donors can opt into SMS notifications (in addition to, or in place of
email notifications) when making donations.

This commit adds `users.phone_number` and
`users.normalized_phone_number` for integration with the [phony_rails](https://github.com/joost/phony_rails)
gem.

### CODE REVIEW

File: app/views/profiles/_profile.html.erb [hunk 1]
```diff
@@ -3,10 +3,19 @@
   <%= inline_svg("email.svg", class: "svg-icon") %>
   <%= profile.email %>
 </span>
+
+<% if profile.phone_number.present? %>
+  <span class="supplier-info__metadata phone-number">
+    <%= inline_svg("location-pin.svg", class: "svg-icon") %>
```
Comment (seanpdoyle) [inline app/views/profiles/_profile.html.erb]: @corwinharrell could you share a suitable SVG for this icon?

File: config/locales/simple_form.en.yml [hunk 2]
```diff
@@ -53,6 +53,7 @@ en:
       profile:
         address: "Street Address for Pickup"
         notes: "Pickup Notes"
+        notes: "Phone Number"
```
Comment (seanpdoyle) [inline config/locales/simple_form.en.yml]: @corwinharrell how should we label this input?

Is `Phone Number` enough, or should we be more explicit -- something like `Phone number to contact about donations`?

File: app/views/profiles/_profile.html.erb [hunk 3]
```diff
@@ -3,10 +3,19 @@
...
+    <%= profile.phone_number.phony_formatted(normalize: :US, spaces: "-") %>
```
Comment (jferris) [inline app/views/profiles/_profile.html.erb]: Does it make sense to extract a method for this directly on `Profile`?

File: app/models/user.rb [hunk 4]
```diff
@@ -14,6 +18,7 @@ class User < ActiveRecord::Base
     message: I18n.t("validations.accepted"),
   }
   validates :password, presence: true, on: :create
+  validates :phone_number, phony_plausible: { normalized_country_code: "US" }
```
Comment (jferris) [inline app/models/user.rb]: Do you need `allow_nil` or `allow_blank` here? What's the expectation for missing phone numbers?

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -8,6 +8,13 @@
   it { should validate_presence_of(:email) }
   it { should validate_presence_of(:password).on(:create) }
 
+  it { should allow_value("(555) 555-5555").for(:phone_number) }
+  it { should allow_value("555-555-5555").for(:phone_number) }
+  it { should allow_value(nil).for(:phone_number) }
+  it { should allow_value("").for(:phone_number) }
```
Comment (seanpdoyle) [inline spec/models/user_spec.rb]: @jferris providing a phone number is entirely optional. 

It doesn't seem to require an `allow_empty` or `allow_nil` option.

File: config/locales/simple_form.en.yml (hunk 2 above)
Comment (corwinharrell) [inline config/locales/simple_form.en.yml]: I think explicit is probably better so we avoid the user wondering what we be using their number for.

File: app/views/profiles/_profile.html.erb (hunk 3 above)
Comment (seanpdoyle) [inline app/views/profiles/_profile.html.erb]: @jferris I like that idea.

**Comment by seanpdoyle** [comment]
@jferris I've pushed some changes -- does this look good?

**Pull request closed without merging**